tests/native_token_bridge.go

```diff
@@ -0,0 +1,382 @@
+package tests
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	nativetokendestination "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/NativeTokenDestination"
+	nativetokensource "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/NativeTokenSource"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	deploymentUtils "github.com/ava-labs/teleporter/utils/deployment-utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func NativeTokenBridge(network network.Network) {
+	const (
+
+		// Each test file needs a unique deployer that must be funded with tokens to deploy
+		deployerKeyStr                     = "aad7440febfc8f9d73a58c3cb1f1754779a566978f9ebffcd4f4698e9b043985"
```

Comment:
I think we can use the funded account of the `network` provider here rather than hard coding keys, right?

We can derive the expected contract addresses using that accounts current nonce, rather than hard coding the first few nonces as well. Same for `erc20_to_native_token_bridge.go`

Reply:
I implemented these tests in a way that they would be parallelizable. If we know we never want to do that, I can change it to use the `fundedKey`.

Reply:
But if we want these to be parallelizable, these files need their own keys so the nonce is independent, and the address can be added to the Native Minter Precompile config.

Reply:
Gotcha, yeah makes sense. @cam-schultz can speak to this better, but I don't think the tests are easily parallelizable as is for the same reason. I have a preference to keep them consistent with each other so that we don't need to handle them on a case-by-case basis moving forward, because if/when we want to make them run in parallel I think having proper locks/synchronization around nonce usage will be the approach rather than trying to use custom accounts for each test run.

Reply:
For all the other tests (other than validator churn), could just generate a random new address, ask the funded address to send some funds, and then every test can act independently. 

The problem with that solution here would be that the key for these tests needs to be known ahead of time to be added to the Native Minter Precompile admin list.

Reply:
Ah, sorry didn't realize the issue with the Native Minter Precompile admin list. Yeah that makes sense to me now. Could you add a comment explaining that a different account is used for these tests because a specific nonce needs to be used to deploy these contracts?